web/controllers/api/docs_controller.ex

```diff
@@ -163,16 +163,13 @@ defmodule HexWeb.API.DocsController do
     HexWeb.CDN.purge_key(:fastly_hexdocs, unversioned_key)
     HexWeb.CDN.purge_key(:fastly_hexdocs, versioned_key)
 
-    # Set docs flag on release
-    HexWeb.Repo.transaction(fn ->
-      Ecto.Changeset.change(release, has_docs: true)
-      |> HexWeb.Repo.update!
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.update(:release, Ecto.Changeset.change(release, has_docs: true))
+      |> Ecto.Multi.update(:package, Ecto.Changeset.change(release.package, docs_updated_at: Ecto.DateTime.utc))
+      |> Ecto.Multi.run(:log, fn _ -> audit(user, "docs.publish", {package, release}) end)
```

Comment:
I'm kinda lost here, could you give me some more guidance how exactly this would work? Sorry for noob questions.

1. What would be the association between Release (or Package or User or ...) and AuditLog? has_one/has_many? Does it matter?

2. Would the association be, sort of, "virtual" and it would be just used for the cast_assoc (or something similar) and the audits table wouldn't directly reference any other table through a foreign key, right? I made the audit_logs schema not directly referencing anything else, and instead we serialize things into `params` JSON column. This could be changed, if desired, since in reality every audited change could be tied to one particular schema like Relase in this particular case.

3. How would usage of audits in controllers look like, as opposed to what I have right now? I think sharing some "wishful code" would be super helpful and it would allow me to work backwards from it.

Anyhow, thanks a lot for the feedback! I agree that if we can just rely on associations (and thus avoid extra DB calls) then it's probably better.